apps/evm/lib/evm/refunds.ex

```diff
@@ -85,18 +86,70 @@ defmodule EVM.Refunds do
 
   # `SSTORE` operations produce a refund when storage is set to zero from some non-zero value.
   def refund(:sstore, [key, new_value], _machine_state, _sub_state, exec_env) do
```

Comment:
Might be out of the scope of this PR/overkill but this is getting pretty complex. Do we want to pull it out into `refunds/sstore.ex` or something?